src/prefect/agent/docker/agent.py

```diff
@@ -442,15 +444,49 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
             "io.prefect.flow-id": flow_run.flow.id,
             "io.prefect.flow-run-id": flow_run.id,
         }
+
+        # Generate a container name to match the flow run name, ensuring it is docker
+        # compatible and unique. Must match `[a-zA-Z0-9][a-zA-Z0-9_.-]+` in the end
+        container_name = slugified_name = (
+            re.sub(
+                # Docker does not allow leading _, -, . so we need to remove those from
+                # the slugified name
+                "^[._-]+",
+                "",
+                slugify(
+                    flow_run.name,
+                    lowercase=False,
+                    # Docker does not limit length but URL limits apply eventually so
+                    # limit the length for safety
+                    max_length=250,
+                    # Docker allows these characters for container names
+                    regex_pattern=r"[^a-zA-Z0-9_.-]+",
+                ),
+            )
+            # Docker does not allow 0 character names so use the flow run id if name
+            # would be empty
+            or flow_run.id
+        )
```

Comment:
This re.sub call could be replaced by a `str.lstrip` call.

```python
slugify(...).lstrip("_.-")
```

Reply:
Ah I considered `lstrip` but thought it took a ordered leading string to remove. I think I'll change that.